src/components/Modal/Modal.tsx

```diff
@@ -64,178 +64,156 @@ export interface ModalProps extends FooterProps {
-        </Header>
-      ) : (
-        <CloseButton
-          onClick={onClose}
-          title={false}
-          testID="ModalCloseButton"
-        />
-      );
+    const content = sectioned
+      ? wrapWithComponent(children, Section, {})
+      : children;
 
-      const labelledBy = title ? this.headerId : undefined;
-
-      dialog = (
-        <Dialog
-          instant={instant}
-          labelledBy={labelledBy}
-          onClose={onClose}
-          onEntered={this.handleEntered}
-          onExited={this.handleExited}
-          large={large}
-          limitHeight={limitHeight}
-        >
-          {headerMarkup}
-          <div className={styles.BodyWrapper}>{bodyMarkup}</div>
-          {footerMarkup}
-        </Dialog>
-      );
+    const body = loading ? (
+      <div className={styles.Spinner}>
+        <Spinner />
+      </div>
+    ) : (
+      content
+    );
 
-      backdrop = <Backdrop />;
-    }
+    const bodyMarkup = src ? (
+      <iframe
+        name={iFrameName}
+        title={iframeTitle}
+        src={src}
+        className={styles.IFrame}
+        onLoad={handleIFrameLoad}
+        style={{height: `${iframeHeight}px`}}
+      />
+    ) : (
+      <Scrollable
+        shadow
+        className={styles.Body}
+        onScrolledToBottom={onScrolledToBottom}
+      >
+        {body}
+      </Scrollable>
+    );
 
-    const animated = !instant;
-
-    return (
-      <WithinContentContext.Provider value>
-        <div ref={this.activatorRef}>{activator}</div>
-        <Portal idPrefix="modal">
-          <TransitionGroup appear={animated} enter={animated} exit={animated}>
-            {dialog}
-          </TransitionGroup>
-          {backdrop}
-        </Portal>
-      </WithinContentContext.Provider>
+    const headerMarkup = title ? (
+      <Header id={headerId} onClose={onClose} testID="ModalHeader">
+        {title}
+      </Header>
+    ) : (
+      <CloseButton onClick={onClose} title={false} testID="ModalCloseButton" />
     );
+
+    const labelledBy = title ? headerId : undefined;
+
+    dialog = (
+      <Dialog
+        instant={instant}
+        labelledBy={labelledBy}
+        onClose={onClose}
+        onEntered={handleEntered}
+        onExited={handleExited}
+        large={large}
+        limitHeight={limitHeight}
+      >
+        {headerMarkup}
+        <div className={styles.BodyWrapper}>{bodyMarkup}</div>
+        {footerMarkup}
+      </Dialog>
+    );
+
+    backdrop = <Backdrop />;
   }
 
-  private handleEntered = () => {
-    const {onTransitionEnd} = this.props;
+  const animated = !instant;
+
+  return (
+    <WithinContentContext.Provider value>
+      <div ref={activatorRef}>{activator}</div>
+      <Portal idPrefix="modal">
+        <TransitionGroup appear={animated} enter={animated} exit={animated}>
+          {dialog}
+        </TransitionGroup>
+        {backdrop}
+      </Portal>
+    </WithinContentContext.Provider>
+  );
+
+  function handleEntered() {
```

Comment:
These functions need to be wrapped in useCallback so we don't trigger rerenders of the elements that they get passed into